cdap-security/src/main/java/io/cdap/cdap/security/authorization/InternalAccessEnforcer.java

```diff
@@ -0,0 +1,126 @@
+/*
+ * Copyright © 2021 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.security.authorization;
+
+import com.google.inject.Inject;
+import io.cdap.cdap.api.security.AccessException;
+import io.cdap.cdap.common.conf.CConfiguration;
+import io.cdap.cdap.common.io.Codec;
+import io.cdap.cdap.proto.element.EntityType;
+import io.cdap.cdap.proto.id.EntityId;
+import io.cdap.cdap.proto.security.Credential;
+import io.cdap.cdap.proto.security.Permission;
+import io.cdap.cdap.proto.security.Principal;
+import io.cdap.cdap.security.auth.AccessToken;
+import io.cdap.cdap.security.auth.InvalidTokenException;
+import io.cdap.cdap.security.auth.TokenManager;
+import io.cdap.cdap.security.auth.UserIdentity;
+import io.cdap.cdap.security.auth.context.SystemAuthenticationContext;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.util.Base64;
+import java.util.Collections;
+import java.util.Set;
+
+public class InternalAccessEnforcer extends AbstractAccessEnforcer {
+  private static final Logger LOG = LoggerFactory.getLogger(InternalAccessEnforcer.class);
+
+  private final TokenManager tokenManager;
+  private final Codec<AccessToken> accessTokenCodec;
+
+  @Inject
+  InternalAccessEnforcer(CConfiguration cConf, TokenManager tokenManager, Codec<AccessToken> accessTokenCodec) {
+    super(cConf);
+    this.tokenManager = tokenManager;
+    this.accessTokenCodec = accessTokenCodec;
+  }
+
+  @Override
+  public void enforce(EntityId entity, Principal principal, Set<? extends Permission> permissions)
+    throws AccessException {
```

Comment:
I prefer wrapping like this more:

```
  public void enforce(EntityId entity, Principal principal, 
                      Set<? extends Permission> permissions) throws AccessException {
```

Same for all other methods in this class

Reply:
Fixed